plugins/bufferlines/bufferline.nix

```diff
@@ -6,359 +6,377 @@
 } @ args:
 with lib; let
   cfg = config.plugins.bufferline;
-  optionWarnings = import ../../lib/option-warnings.nix args;
   helpers = import ../helpers.nix args;
 
-  highlight = mkOption {
-    type = types.nullOr (types.submodule ({...}: {
-      options = {
-        guifg = mkOption {
-          type = types.nullOr types.str;
-          description = "foreground color";
-          default = null;
-        };
-        guibg = mkOption {
-          type = types.nullOr types.str;
-          description = "background color";
-          default = null;
-        };
-      };
-    }));
-    default = {};
+  highlight = helpers.mkCompositeOption "Highlight option" {
+    guifg = helpers.mkNullOrOption types.str "foreground color";
+
+    guibg = helpers.mkNullOrOption types.str "background color";
   };
 in {
-  imports = [
-    (optionWarnings.mkDeprecatedOption {
-      option = ["plugins" "bufferline" "indicatorIcon"];
-      alternative = ["plugins" "bufferline" "indicator" "icon"];
-    })
-  ];
-
```

Comment:
A reason for having deleted the warning ? Or was it deleted upstream ?

Reply:
Because `mkDeprecatedOption` does not work with options that are nested in `submodules` (i.e. options inside `mkCompositOption` options).

Reply:
Then we need to find a way to make it work with submodules, we can't break the options of the users like this, we need to be forgiving to deprecated names. We also should add the date to the introduction of the option, and delete it maybe something like after 2 months/ 1.5 month I'm not really sure

Reply:
Yes, you are completely right.

Reply:
Fixed :)